src/Refactoring-Core/RBExtractToTemporaryRefactoring.class.st

```diff
@@ -63,7 +63,7 @@ RBExtractToTemporaryRefactoring >> compileNewMethod [
 RBExtractToTemporaryRefactoring >> constructAssignmentFrom: aNode [
 	| valueNode |
 	valueNode := RBVariableNode named: newVariableName.
-	^RBAssignmentNode variable: valueNode value: aNode
+	^RBAssignmentNode variable: valueNode value: aNode copy.
```

Comment:
That is a case of defensive programming, and perhaps it's overzealous. I like being able to rely on the `parent` backlink of parse nodes to be accurate, and TBH I really wish they were immutable after creation/copy-on-write. In this particular case I'm 99% sure the code will work without it, but note that if (for whatever reason) it were refactored so that `constructAssignmentFrom:` were called prior to retrieving the `statementNode`, it would fail because the parent backlink would be gone. I would be okay with removing the `#copy` though...on reflection I can see the argument that, though it might be nice to be more careful with the parent backlinks, that would be something better done all at once separate from other fixes.

(I was also wondering if there might not be a risk of modifying the One True Parse Tree of the method, the one that's cached and used for things like breakpoints, but I see that that is accessed through a separate selector `#ast` that nobody writing refactoring code is likely to think of using—so that's good.)

Reply:
I get your point, and I'm all in for defensive programming. I like the idea that these defensive features are introduced in a separate PR where they would also be explained with a comment saying what its defending from. Because my first reaction was: "why we need copy here?" and a comment would helped a lot.
I'm fine moving `#copy` to another PR, but I'm also fine with staying in this one, and in that case I would kindly ask you to put a small comment :)

Reply:
That makes sense, yeah—I'll do it separately in the future, for now to keep things simple I added comments (well, the same one in both locations).